src/SourceBuild/content/eng/tools/PrBaselinePublisher/Program.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.CommandLine;
+using Microsoft.Extensions.Logging;
+
+namespace PrBaselinePublisher;
+
+public class Program
+{
+    public static readonly CliArgument<string> UpdatedTestsResultsPath = new("updated-test-results-path")
+    {
+        Description = "The directory containing the updated test files published by the associated test.",
+        Arity = ArgumentArity.ExactlyOne
+    };
+
+    public static readonly CliArgument<int> BuildId = new("build-id")
+    {
+        Description = "The id of the build that published the updated test files.",
+        Arity = ArgumentArity.ExactlyOne
+    };
+
+    public static readonly CliOption<string> Title = new("--title", "-t")
+    {
+        Description = "The title of the PR.",
+        Arity = ArgumentArity.ZeroOrOne,
+        DefaultValueFactory = _ => "Update Test Baselines and Exclusions"
+    };
+
+    public static readonly CliOption<string> Branch = new("--branch", "-b")
+    {
+        Description = "The target branch of the PR.",
+        Arity = ArgumentArity.ZeroOrOne,
+        DefaultValueFactory = _ => "main"
+    };
+
+    public static readonly CliOption<string> Repo = new("--repo", "-r")
+    {
+        Description = "The repository to create the PR in.",
+        Arity = ArgumentArity.ZeroOrOne,
+        DefaultValueFactory = _ => "dotnet/installer"
+    };
+
+    public static readonly CliOption<string> OriginalTestResultsPath = new("--original-test-results-path", "-o")
+    {
+        Description = "The directory where the original test files are located.",
+        Arity = ArgumentArity.ExactlyOne,
+        DefaultValueFactory = _ => "src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets"
+    };
```

Comment:
We should avoid defining a default value for these two options. They should be explicitly passed in by the pipeline. I'd like to have this tool as generalized as possible to allow it to potentially be used by the product repos (assuming we can push license detection at that level).